contracts/rollup/Rollup.sol

```diff
@@ -279,9 +289,11 @@ contract Rollup is BatchManager, EIP712, IEIP712 {
     }
 
     function submitDeposits(
+        uint256 batchID,
         Types.CommitmentInclusionProof memory previous,
         Types.SubtreeVacancyProof memory vacant
     ) public payable onlyCoordinator isNotRollingBack {
+        validateBatchID(batchID);
         uint256 preBatchID = nextBatchID - 1;
```

Comment:
Since we have established that nextBatchID== batchID, we can use batchID here for cheaper gas.
```suggestion
        uint256 preBatchID = batchID - 1;
```